client/state/selectors/is-site-checklist-complete.js

```diff
@@ -6,6 +6,8 @@
 import getSiteChecklist from 'state/selectors/get-site-checklist';
 import isSiteChecklistLoading from 'state/selectors/is-site-checklist-loading';
 import { getSiteFrontPage } from 'state/sites/selectors';
+import { getTaskList } from 'my-sites/checklist/wpcom-checklist/wpcom-task-list';
+import { getTaskUrls } from 'my-sites/checklist/wpcom-checklist/component';
```

Comment:
This import is making a selector depend on a React component. That's not ideal, as now state is strongly coupled to presentation, and pulling in a whole bunch of stuff. Don't trust tree-shaking to work correctly 🙂 

If you want to reuse the selector that was added to `wpcom-checklist/component.jsx`, I would suggest extracting it to its own file somewhere in state, and reusing it in both this file and the checklist component. That will make the dependency graph much cleaner.

Reply:
Thanks @sgomes! I'll do that. I think there might be further dependencies to unravel there to.

Reply:
Thank you for making the change! This is looking a lot better now, and the bundle size changes are no longer an issue 👍